parts/windows/kuberneteswindowssetup.ps1

```diff
@@ -502,16 +562,29 @@ finally
 
     Logs-To-Event -TaskName "AKS.WindowsCSE.cse_main" -TaskMessage "ExitCode: $global:ExitCode. ErrorMessage: $global:ErrorMessage."
 
-    # $CSEResultFilePath is used to avoid running CSE multiple times
-    if ($global:ExitCode -ne 0) {
-        # $JsonString = "ExitCode: |{0}|, Output: |{1}|, Error: |{2}|"
-        # Max length of the full error message returned by Windows CSE is ~256. We use 240 to be safe.
-        $errorMessageLength = "ExitCode: |$global:ExitCode|, Output: |$($global:ErrorCodeNames[$global:ExitCode])|, Error: ||".Length
-        $turncatedErrorMessage = $global:ErrorMessage.Substring(0, [Math]::Min(240 - $errorMessageLength, $global:ErrorMessage.Length))
-        Set-Content -Path $CSEResultFilePath -Value "ExitCode: |$global:ExitCode|, Output: |$($global:ErrorCodeNames[$global:ExitCode])|, Error: |$turncatedErrorMessage|"
-    }
-    else {
-        Set-Content -Path $CSEResultFilePath -Value $global:ExitCode -Force
+    # Create appropriate completion file based on mode
+    if ($PreProvisionOnly) {
```

Comment:
Apart for the file name (provision.complete and preprovision.complete), these two branches appear the same. Can we just set the file name in the if branch then have the writing to file as a common block? That'll save some bytes (CSE files are size constrained) and mean they'll stay in sync in future.

Reply:
done